src/cpu/gemm_convolution_utils.cpp

```diff
@@ -444,18 +444,41 @@ void im2col(const conv_gemm_conf_t &jcp, const data_type_t *__restrict im,
                             const int ow_end
                                     = (oh == last_oh) ? (last_ow + 1) : jcp.ow;
                             data_t *__restrict col_ = col_k + oh * jcp.ow - ss;
-                            if (ih < 0 || ih >= jcp.ih)
+                            if (ih < 0 || ih >= jcp.ih) {
+                                /*
                                 for (int ow = ow_begin; ow < ow_end; ow++)
                                     col_[ow] = zero_val;
-                            else {
-                                for (int ow = ow_begin; ow < ow_end; ++ow) {
+                                */
+                                memset(col_ + ow_begin, zero_val,
+                                        (ow_end - ow_begin) * sizeof(data_t));
+                            } else {
+                                auto iw_begin = ow_begin;
+                                auto iw_end = ow_end;
+                                auto left_0s
+                                        = std::max(0, lp - kw * dw - iw_begin);
+                                auto right_0s = std::max(
+                                        0, iw_end - jcp.iw - lp + kw * dw);
+                                auto len = ow_end - ow_begin;
+                                // content length
+                                auto c_len = len - left_0s - right_0s;
+
```

Comment:
(random place) please drop `x64: ` from the commit message as the patch seems to be about the common code